client/executor/wasmtime/src/tests.rs

```diff
@@ -168,24 +178,36 @@ fn test_stack_depth_reaching() {
 }
 
 #[test]
-fn test_max_memory_pages_imported_memory() {
-	test_max_memory_pages(true);
+fn test_max_memory_pages_imported_memory_without_precompilation() {
+	test_max_memory_pages(true, false);
+}
+
+#[test]
+fn test_max_memory_pages_exported_memory_without_precompilation() {
+	test_max_memory_pages(false, false);
+}
+
+#[test]
+fn test_max_memory_pages_imported_memory_with_precompilation() {
+	test_max_memory_pages(true, true);
 }
 
 #[test]
-fn test_max_memory_pages_exported_memory() {
-	test_max_memory_pages(false);
+fn test_max_memory_pages_exported_memory_with_precompilation() {
+	test_max_memory_pages(false, true);
 }
 
-fn test_max_memory_pages(import_memory: bool) {
+fn test_max_memory_pages(import_memory: bool, precompile_runtime: bool) {
 	fn try_instantiate(
 		max_memory_size: Option<usize>,
 		wat: String,
+		precompile_runtime: bool,
 	) -> Result<(), Box<dyn std::error::Error>> {
 		let runtime = {
 			let mut builder = RuntimeBuilder::new_on_demand();
 			builder.use_wat(wat);
 			builder.max_memory_size(max_memory_size);
+			builder.precompile_runtime(precompile_runtime);
 			builder.build()
 		};
```

Comment:
I know it was not you who introduced this builder, but the best practice for the [builder pattern in Rust](https://rust-unofficial.github.io/patterns/patterns/creational/builder.html) is that each method takes `self` and returns it back. That allows one to write:

```rust
let runtime = RuntimeBuilder::new_on_demand()
	.use_wat(wat)
	.max_memory_size(max_memory_size)
	.precompile_runtime(precompile_runtime)
	.build();
```

Reply:
Okay, you can use fluent API like I wrote with `&mut self` arguments, too.